app/main/views/questionnaire.py

```diff
@@ -371,4 +371,16 @@ def _render_template(context, group_id=None, group_instance=0, block_id=None, te
     return render_theme_template(theme, template, meta=metadata_context,
                                  content=context,
                                  previous_location=previous_url,
+                                 navigation=front_end_navigation,
                                  schema=g.schema_json)
+
+
+def get_front_end_navigation(group_id, group_instance, navigator):
```

Comment:
Can the navigator call be updated to make the schema check and return this list, so this method isn't necessary?